src/api/c++/z3++.h

```diff
@@ -472,13 +539,20 @@ namespace z3 {
     public:
         params(context & c):object(c) { m_params = Z3_mk_params(c); Z3_params_inc_ref(ctx(), m_params); }
         params(params const & s):object(s), m_params(s.m_params) { Z3_params_inc_ref(ctx(), m_params); }
-        ~params() { Z3_params_dec_ref(ctx(), m_params); }
+        params(params && s) noexcept : object{std::forward<object>(s)}, m_params{s.m_params} { s.m_params = nullptr; }
```

Comment:
the object bit is redundant

Reply:
Responded to in a different comment with the same message.